murenn/dtcwt/utils.py

```diff
@@ -0,0 +1,63 @@
+import numpy as np
+import torch
+
+
+def reflect(x, minx, maxx):
+    """Reflect the values in matrix *x* about the scalar values *minx* and
+    *maxx*.  Hence a vector *x* containing a long linearly increasing series is
+    converted into a waveform which ramps linearly up and down between *minx*
+    and *maxx*.  If *x* contains integers and *minx* and *maxx* are (integers +
+    0.5), the ramps will have repeated max and min samples.
+
+    .. codeauthor:: Rich Wareham <rjw57@cantab.net>, Aug 2013
```

Comment:
add yourself as codeauthor and say "adapted from:" then cite Wareham and Kingsbury

Reply:
done

Reply:
https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-codeauthor
for your information codeauthor:: is a Sphinx directive